pkg/snapshot/snapshot.go

```diff
@@ -179,16 +178,19 @@ func (s *Snapshotter) scanFullFilesystem() ([]string, []string, error) {
 			return nil, nil, fmt.Errorf("could not check if file has changed %s %s", path, err)
 		}
 		if fileChanged {
-			logrus.Tracef("Adding %s to layer, because it was changed.", path)
-			filesToAdd = append(filesToAdd, path)
+			files, err := filesWithLinks(path)
+			if err != nil {
+				return nil, nil, err
+			}
+			logrus.Debugf("Adding files %s to layer, because it was changed.", files)
+			filesToAdd = append(filesToAdd, files...)
```

Comment:
Why add both the link and fully resolved path?

Reply:
We need to save both symlink and target file to kaniko workspace if symlink is referenced across stages. 
If we don't then the symlink points to a deadlink and results into an error.

Reply:
Might worth adding a comment. Your call